src/borg/archive.py

```diff
@@ -91,6 +91,7 @@ def show_progress(self, item=None, final=False, stream=None, dt=None):
                 space = columns - swidth(msg)
                 if space < swidth('...') + swidth(path):
                     path = '%s...%s' % (path[:(space // 2) - swidth('...')], path[-space // 2:])
+                space -= swidth(path)
                 msg += "{0:<{space}}".format(path, space=space)
```

Comment:
> "<N" means "left-align in a N spaces wide field", not "add N spaces after the argument".

Correct. But this calculation works in terms of characters, not terminal cells.

Example: space is 20. The path contains 8 characters. Therefore 12 spaces are appended. But wait, the 8 characters from the path are actually two-cell characters. Now the resulting string occupies 16 + 12 = 28 cells, not 20.

Therefore we subtract the cell-count of the path from the space to pad width. space is 20, the path contains 8 double-cell characters, then space is only 20 - 16 = 4. Resulting string is 20 cells wide.

Could also write "msg += path + ' ' * space" to make it a bit clearer.